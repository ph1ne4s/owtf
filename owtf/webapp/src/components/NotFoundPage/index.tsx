/*
 * Component to show if page not found.
 */
import React from "react";
import { FaRobot } from "react-icons/fa";
import { Link } from "react-router-dom";

export default class NotFoundPage extends React.Component {
  render() {
    return (
      <section className="notFoundPage">
        <div className="notFoundPage__contentWrapper">
          <div className="notFoundPage__contentWrapper__headingAndImageContainer">
            <FaRobot />
            <h2>404</h2>
          </div>

          <div className="notFoundPage__contentWrapper__errorInfoContainer">
            <h3>Page Not Found!</h3>
            <p>We are unable to find the page you're looking for.</p>
          </div>

          <div className="notFoundPage__contentWrapper__buttonContainer">
            <Link to="/">Back to Home Page</Link>
          </div>
        </div>
      </section>
    );
  }
}
