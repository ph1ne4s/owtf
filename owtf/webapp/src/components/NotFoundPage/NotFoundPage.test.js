import React from "react";
import { shallow } from "enzyme";
import NotFoundPage from "./index";
import "../../setupTests";
import toJson from "enzyme-to-json";

const setUp = (props = {}) => {
  const component = shallow(<NotFoundPage {...props} />);
  return component;
};

describe("NotFoundPage Component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("Should render without errors", () => {
    const tree = toJson(component);
    expect(component.find("h3").length).toBe(1);
    expect(component.find("h3").text()).toBe("Page Not Found!");
    expect(tree).toMatchSnapshot();
  });
});
