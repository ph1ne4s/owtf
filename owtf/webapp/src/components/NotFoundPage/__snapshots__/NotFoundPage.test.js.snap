// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component Should render without errors 1`] = `
<section
  className="notFoundPage"
>
  <div
    className="notFoundPage__contentWrapper"
  >
    <div
      className="notFoundPage__contentWrapper__headingAndImageContainer"
    >
      <FaRobot />
      <h2>
        404
      </h2>
    </div>
    <div
      className="notFoundPage__contentWrapper__errorInfoContainer"
    >
      <h3>
        Page Not Found!
      </h3>
      <p>
        We are unable to find the page you're looking for.
      </p>
    </div>
    <div
      className="notFoundPage__contentWrapper__buttonContainer"
    >
      <Link
        replace={false}
        to="/"
      >
        Back to Home Page
      </Link>
    </div>
  </div>
</section>
`;
